### PR TITLE
feat: Add Elementary's recommended editorconfig

### PR DESCRIPTION
**Pull request by devnix**

### CODE REVIEW

**Comment by nahuelwexd** [comment]
I don't know if we should keep `max_line_length`, since apparently it was removed from the EditorConfig specification.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+root = true
+
+# elementary defaults
+[*]
+charset = utf-8
+end_of_line = lf
+indent_size = tab
+indent_style = space
+insert_final_newline = true
+max_line_length = 80
```
Comment (nahuelwexd) [inline .editorconfig:11]: I don't know if we should keep `max_line_length`, since apparently it was removed from the EditorConfig specification.

**Review by nahuelwexd — CHANGES_REQUESTED** [review]

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+tab_width = 4
+​
+# Markup files
+[{*.html,*.xml,*.xml.in,*.yml}]
```
Comment (dar5hak) [inline .editorconfig:15]: Shall we add JSON here too?

File: .editorconfig (hunk 1 above)
Comment (devnix) [reply .editorconfig:11]: You're totally right

File: .editorconfig (hunk 2 above)
Comment (devnix) [reply .editorconfig:15]: As a 2 space tab width?

**Review by nahuelwexd — APPROVED** [review]

File: .editorconfig (hunk 2 above)
Comment (dar5hak) [reply .editorconfig:14]: Yeah

Comment (dar5hak) [reply .editorconfig:14]: I'll go ahead and make the change, then merge this.

Comment (dar5hak) [reply .editorconfig:14]:
```suggestion
[{*.html,*.xml,*.xml.in,*.yml,*.json}]
```

**Comment by devnix** [comment]
I would personally default json files (and specially `package.json` file) to 4 spaces, as it seems to be a de-facto standard around there (and it does not have more than a couple of indentation levels)

**Comment by dar5hak** [comment]
> I would personally default json files (and specially package.json file) to 4 spaces, as it seems to be a de-facto standard around there (and it does not have more than a couple of indentation levels)

Fair enough. Can you go ahead and change that?

Then I'll squash with a lowercase, present tense commit message because that's what the examples here use: https://www.conventionalcommits.org/en/v1.0.0/

**Comment by devnix** [comment]
@dar5hak It's ok like that?

**Comment by dar5hak** [comment]
I lowercased everything, including elementary because that's how they stylize it, but otherwise, all okay.

**Comment by devnix** [comment]
Nice! The spec mentions this:

>Q: Are the types in the commit title uppercase or lowercase?
>A: Any casing may be used, but it’s best to be consistent.

But I'm ok with any option, I'll stick to lowercase then!

**Pull request merged**